### PR TITLE
PvlKeyword from JSON

### PR DESCRIPTION
**Pull request by acpaquette**
Allows PvlKeywords to have values set and added from JSON objects

## Description
Adds two new methods to the PvlKeyword interface, setJsonValue and addJsonValue that allows PvlKeyword values to be set and added to from JSON objects, respectively

## Related Issue
N/A

## Motivation and Context
Bugfix from working with ALE ISDs.

## How Has This Been Tested?
Hand tested with ALE ISDs, no other tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/PvlKeyword/PvlKeyword.cpp [hunk 1]
```diff
@@ -157,6 +158,26 @@ namespace Isis {
...
 
+  /**
+   * Sets new value from Json.
+   *
+   * If no current value exists, this method sets the given json value
+   * to the PvlKeyword.  Otherwise, it clears any existing values
+   * and resets to the value given using addJsonValue(). Defaults to
+   * unit = "" (empty QString).
+   *
+   * @param value New value to be assigned.
```
Comment (jessemapel) [inline isis/src/base/objs/PvlKeyword/PvlKeyword.cpp]: Update for jsonobj

File: isis/src/base/objs/PvlKeyword/PvlKeyword.cpp [hunk 2]
```diff
@@ -157,6 +158,26 @@ namespace Isis {
...
+   * @param unit Units of measurement corresponding to the value.
+   *
+   * @see addValue()
```
Comment (jessemapel) [inline isis/src/base/objs/PvlKeyword/PvlKeyword.cpp]: see addJsonValue

File: isis/src/base/objs/PvlKeyword/PvlKeyword.cpp [hunk 3]
```diff
@@ -267,6 +288,42 @@ namespace Isis {
...
 
+  /**
+   * Adds a value with units.
+   *
+   * If no current value exists, this method sets the given json value.
+   * Otherwise, it retains any current values and adds the json value
+   * given to the array of values for this PvlKeyword object using addValue.
+   * Defaults to unit = "" (empty QString).
+   *
+   * @param value New value to be assigned.
```
Comment (jessemapel) [inline isis/src/base/objs/PvlKeyword/PvlKeyword.cpp]: Same as above, fix param name

File: isis/src/base/objs/PvlKeyword/PvlKeyword.cpp [hunk 4]
```diff
@@ -267,6 +288,42 @@ namespace Isis {
...
+    }
+    else if (jsonobj.is_boolean())
+    {
+      value = QString(jsonobj.get<bool>() ? "true" : "false");
+    }
+    else if (jsonobj.is_null())
+    {
+      value = QString("Null");
+    }
+    else
```
Comment (jessemapel) [inline isis/src/base/objs/PvlKeyword/PvlKeyword.cpp:323]: What about arrays?

**Review by jessemapel — COMMENTED** [review]
This needs a unit test for the PvlKeyword stuff. You should also replace the stuff in the PvlObject constructor with this.

File: isis/src/base/objs/PvlObject/PvlObject.cpp [hunk 5]
```diff
@@ -63,27 +63,14 @@ namespace Isis {
...
-        else if(it.value().is_null()) {
-          keyword.setName(QString::fromStdString(it.key()));
-          keyword.setValue(QString("Null"));
-        }
         else {
-          keyword.setName(QString::fromStdString(it.key()));
-          keyword.setValue(QString::fromStdString(it.value()));
+          keyword.setJsonValue(*it);
         }
         addKeyword(keyword);
```
Comment (acpaquette) [inline isis/src/base/objs/PvlObject/PvlObject.cpp]: @jessemapel  When you refer to replacing the constructor is this what you are referring to?

File: isis/src/base/objs/PvlKeyword/PvlKeyword.cpp (hunk 4 above)
Comment (acpaquette) [reply isis/src/base/objs/PvlKeyword/PvlKeyword.cpp:323]: So far arrays need to be handled by the `PvlObject` which is happening in this PR

File: isis/src/base/objs/PvlObject/PvlObject.cpp (hunk 5 above)
Comment (jessemapel) [reply isis/src/base/objs/PvlObject/PvlObject.cpp]: It is, I somehow smashed this together with the PvlKeyword functions and thought this was a separate function.

File: isis/src/base/objs/PvlKeyword/PvlKeyword.cpp (hunk 4 above)
Comment (acpaquette) [reply isis/src/base/objs/PvlKeyword/PvlKeyword.cpp:323]: Discussed offline and we are going to throw an error message if there is a json object that contains a 2+D array

**Review by jessemapel — APPROVED** [review]

**Pull request merged**